src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -3369,13 +3369,10 @@ void G1CollectedHeap::fill_with_dummy_object(HeapWord* start, HeapWord* end, boo
 
 void G1CollectedHeap::start_codecache_marking_cycle_if_inactive() {
   if (!CodeCache::is_gc_marking_cycle_active()) {
-    // This is the normal case when we do not call collect when a
-    // concurrent mark is ongoing. We then start a new code marking
-    // cycle. If, on the other hand, a concurrent mark is ongoing, we
-    // will be conservative and use the last code marking cycle. Code
-    // caches marked between the two concurrent marks will live a bit
-    // longer than needed.
```

Comment:
I removed the comment because I found it a little missleading. If "concurrent marking" is G1 cm then we never reach here with an ongoing concurrent marking. Reaching here G1 cm was either finished normally, undone or aborted. But maybe "concurrent marking" means marking of nmethods?